src/http/mod.rs

```diff
@@ -156,11 +172,22 @@ impl HttpClient {
                 },
                 Err(e) => {
                     metrics::DOCUMENT.with_label_values(&["fetch_error"]).inc();
-                    error!(
-                        "Unable to fetch document, id={}, reason={}, url={}",
-                        req_id, e, url
-                    );
-                    Err(Errors::FetchFailed)
+                    let err = e
+                        .source()
+                        .unwrap()
+                        .downcast_ref::<std::io::Error>()
+                        .unwrap();
```

Comment:
I this result should be handled properly instead of unwrapping.

Reply:
yes! it was a bug , I have handled it better in my next commit